app/javascript/packs/form-validation.js

```diff
@@ -19,25 +19,49 @@ function disableFormSubmit(event) {
   });
 }
 
+function resetInput(input) {
+  input.setCustomValidity('');
+  input.setAttribute('aria-invalid', 'false');
+  input.classList.remove('usa-input--error');
+  const errorMessages = input.parentNode?.querySelectorAll('.invalid-inline');
+  errorMessages.forEach((message) => {
+    message.classList.add('display-none');
+  });
+}
+
+function toggleErrorForUnit(input) {
+  if (input.validity.valueMissing) {
+    const errorMessage = input.parentNode?.querySelector('.input-required');
+    errorMessage.classList.remove('display-none');
+  } else if (input.validity.patternMismatch) {
+    const errorMessage = input.parentNode?.querySelector('.pattern-mismatch');
+    errorMessage.classList.remove('display-none');
+  }
```

Comment:
To avoid having to implement logic and class toggling for each validation error, I wonder if we might be able to extend what we're doing with `.display-if-invalid` using [the different possible validation errors](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#the-constraint-validation-api:validitystate-3), and take advantage of the fact that [`aria-invalid` can be any arbitrary string](https://www.w3.org/TR/wai-aria-1.1/#aria-invalid).

For example: https://codepen.io/aduth/pen/LYLvyOb 

Thoughts?